src/main/java/org/kairosdb/rollup/RollUpJob.java

```diff
@@ -122,21 +120,14 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
 					}
 					finally
 					{
-						try
-						{
-							ThreadReporter.setReportTime(System.currentTimeMillis());
-							ThreadReporter.clearTags();
-							ThreadReporter.addTag("host", hostName);
-							ThreadReporter.addTag("rollup", rollup.getSaveAs());
-							ThreadReporter.addTag("rollup-task", task.getName());
-							ThreadReporter.addTag("status", success ? "success" : "failure");
-							ThreadReporter.addDataPoint(ROLLUP_TIME, System.currentTimeMillis() - ThreadReporter.getReportTime());
-							ThreadReporter.submitData(longDataPointFactory, stringDataPointFactory, publisher);
-						}
-						catch (DatastoreException e)
-						{
-							log.error("Could not report metrics for rollup job.", e);
-						}
+						ThreadReporter.setReportTime(System.currentTimeMillis());
+						ThreadReporter.clearTags();
+						ThreadReporter.addTag("host", hostName);
+						ThreadReporter.addTag("rollup", rollup.getSaveAs());
+						ThreadReporter.addTag("rollup-task", task.getName());
+						ThreadReporter.addTag("status", success ? "success" : "failure");
+						ThreadReporter.addDataPoint(ROLLUP_TIME, System.currentTimeMillis() - ThreadReporter.getReportTime());
+						ThreadReporter.submitData(longDataPointFactory, publisher);
```

Comment:
Was ThreadReporter.submitData the only thing that could throw and that's why you removed the try?

Reply:
Yep. It will change even more radically when I convert the recording layer to use ISM metrics client so we can plug V2 into it as a sink.